### DISCUSSION TITLE
"All Sampling methods" option

### DISCUSSION THREAD

**Discussion by marcelo-mason** [hlky/stable-diffusion-webui#1242] [Feature suggestions]
An option to run all sampling methods with current configuration at the same time, have them output like a batch.

![image](https://user-images.githubusercontent.com/577340/191377900-7a405dde-b8f7-4aee-877e-39167dea60e9.png)

**Comment by protoplm** [comment]
I'm working on something that can do this in streamlit. I've got the concept laid out using cfg_scale and some outside mockup of loading them to a grid to get a comparison. Currently trying to expand it to the things I would find interesting to iterate through. So far I have:

-model.ckpt
-cfg_scale
-sampling steps
-sampler

![test_output](https://user-images.githubusercontent.com/57930981/191385133-994def76-8271-4d65-969a-2a314ccf771a.png)

I would like to display the output in a grid like this to the UI, and also place it into the folder along with the individual images.

**Reply by yourjelly** [reply]
A prompt matrix of cfg and sampling is something I would like to see for sure.

**Comment by protoplm** [comment]
Works well locally. The implementation isn't up to standards for the repo though, so it will take some time to have available.

Potentially useful for https://github.com/sd-webui/stable-diffusion-webui/discussions/178

![sampler_test](https://user-images.githubusercontent.com/57930981/191604130-4cacdefa-cd92-4ba8-a263-f71039a298dc.png)
